src/script/lua_api/l_mainmenu.cpp

```diff
@@ -618,16 +618,17 @@ int ModApiMainMenu::l_create_world(lua_State *L)
 
 	std::string path = porting::path_user + DIR_DELIM
 			"worlds" + DIR_DELIM
-			+ name;
+			+ sanitizeDirName(name, "world_");
 
 	std::vector<SubgameSpec> games = getAvailableGames();
 
 	if ((gameidx >= 0) &&
 			(gameidx < (int) games.size())) {
 
 		// Create world if it doesn't exist
-		if (!loadGameConfAndInitWorld(path, games[gameidx])) {
-			lua_pushstring(L, "Failed to initialize world");
+		std::string error;
+		if (!loadGameConfAndInitWorld(path, name, games[gameidx], true, &error)) {
+			lua_pushstring(L, std::string("Failed to initialize world: " + error).c_str());
```

Comment:
the string construction here is redundant, you probably meant 

```c++
(std::string("Failed to initialize world: ") + error).c_str()
```

which can be simplified to

```cpp
("Failed to initialize world: "s + error).c_str()
```

But given the code in your PR compiles, this is somehow OK:

```cpp
("Failed to initialize world: " + error).c_str()
```

Reply:
Fixed with the first option. The `""s` notation doesn't compile on my end for whatever reason, but you're correct that I missed those parens.